csrc/domain_view.h

```diff
@@ -0,0 +1,137 @@
+// clang-format off
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2023-present NVIDIA CORPORATION & AFFILIATES.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+// clang-format on
+#pragma once
+
+#include <ir/builder.h>
+#include <ir/internal_base_nodes.h>
+#include <val_graph.h>
+
+#ifndef DYNAMIC_TYPE_CHECK
+#define DYNAMIC_TYPE_CHECK NVF_ERROR
+#endif
+
+#include <dynamic_type/dynamic_type.h>
+
+// IterDomainLikeObjectView, or IDLOView in short, is a convenient helper class
```

Comment:
What's the motivation of allowing just `IterDomain*` and a group of `IterDomain*`?

Reply:
I think it is mostly for "I want to merge this single ValGroup with multiple ValGroups separately". For example, if I want to manually create an exact graph, dim0 of all tensors are mapped together, dim1 are not because some are broadcasting some are concrete. To schedule, likely I will need to merge this single dim0 with all these different dim1s. For the IterDomain*, it is less interesting, but why not :-D.